File: logisim/app/src/main/java/logisim/gui/main/AttrTableToolModel.java
/* Copyright (c) 2011, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.gui.main;

import logisim.data.Attribute;
import logisim.gui.generic.AttributeSetTableModel;
import logisim.proj.Project;
import logisim.tools.Tool;

public class AttrTableToolModel extends AttributeSetTableModel {
	Project proj;
	Tool tool;
	
	public AttrTableToolModel(Project proj, Tool tool) {
		super(tool.getAttributeSet());
		this.proj = proj;
		this.tool = tool;
	}
	
	@Override
	public String getTitle() {
		return Strings.get("toolAttrTitle", tool.getDisplayName());
	}
	
	public Tool getTool() {
		return tool;
	}
	
	@Override
	public void setValueRequested(Attribute<Object> attr, Object value) {
		proj.doAction(ToolAttributeAction.create(tool, attr, value));
	}
}

File: LogisimPlus/LogisimPlus/logisim/gui/main/AttrTableToolModel.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

/* Copyright (c) 2011, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.gui.main
{
	using logisim.data;
	using AttributeSetTableModel = logisim.gui.generic.AttributeSetTableModel;
	using Project = logisim.proj.Project;
	using Tool = logisim.tools.Tool;

	public class AttrTableToolModel : AttributeSetTableModel
	{
		internal Project proj;
		internal Tool tool;

		public AttrTableToolModel(Project proj, Tool tool) : base(tool.AttributeSet)
		{
			this.proj = proj;
			this.tool = tool;
		}

		public override string Title
		{
			get
			{
				return Strings.get("toolAttrTitle", tool.DisplayName);
			}
		}

		public virtual Tool Tool
		{
			get
			{
				return tool;
			}
		}

		public override void setValueRequested(Attribute<object> attr, object value)
		{
			proj.doAction(ToolAttributeAction.create(tool, attr, value));
		}
	}

}
